topic.go

```diff
@@ -184,6 +290,23 @@ func WithReadiness(ready RouterReady) PubOpt {
 	}
 }
 
+// WithWaitUntilQueued blocks the publish until the message has been added to the outbound message queue for
+// as many peers as the arg indicates
+// A value of -1 means all peers in mesh/fanout for gossipsub & all subscribed peers in floodsub
+// Please note that if nPeers is -1, the behavior is not fail fast
```

Comment:
Does a "non fail fast" behaviour make sense here ? If the caller is willing to wait for the message to be added to the queues for all peers, I figured we might as well wait till the end so we can log all failures for better debugging later.